src/lib/ngsi/ContextAttributeVector.cpp

```diff
@@ -99,16 +98,20 @@ std::string ContextAttributeVector::toJsonTypes()
 
     out += JSON_STR(attrName) + ":{" + JSON_STR("types") + ":[";
 
-    std::map<std::string, int>::iterator jt;
-    unsigned int                         jx;
-    for (jt = attrTypes.begin(), jx = 0; jt != attrTypes.end(); ++jt, ++jx)
+    if (noAttrDetail == false)
     {
-      std::string type = jt->first;
-      out += JSON_STR(type);
+      std::map<std::string, int>::iterator jt;
```

Comment:
ok, I understand the 'strategy'.
It makes the function a bit hard to understand though.
If we do this, it must be clearly documented in the code